### ISSUE TITLE
Incompatibility with Broadwell SMBIOS, works with Haswell on 5500U.

### ISSUE THREAD

**Issue by thePsguy** [Piker-Alpha/ssdtPRGen.sh#186]
When using a Broadwell SMBIOS, extracted from a real MacBookAir7,2, the CPU (5500U), never goes below 1.3GHz. I have tried other Broadwell SMBIOS as well.
Using older SMBIOS like MacBookAir6,2, it complains of incompatible board-id, but goes down to 800Mhz, with the generated ssdt.
The SMBIOS seem to be perfect and error free.
It is supposed to go further below 800MHz according to ark.intel, which it does not in  any case.

**Comment by shixuev5** [comment]
Yes I have the same problem when using Broadwell SMBIOS, The cpu never below 1.3GHz. Wait for fix it.Thanks

**Comment by Piker-Alpha** [comment]
The CPUFloor frequency can be found in the FrequencyVector data in Mac-937CB26E2E02BB01.plist The second or third value is set to 0x0d000000 (sorry cannot recall which one it is) and when you change that, say to 0x08000000, then it will throttle down to 800MHz.

Edit: You can also use the FrequencyData (item 0) of a plist that works for your setup. Toleda confirmed this to work on his Broadwell setup with an Intel i5-5675c Adding the data to a dummy kext is likely the best solution. Or patching it with Clover maybe?

**Comment by shixuev5** [comment]
The data in Mac-937CB26E2E02BB01.plist is like that AgAAAA0AAAABAAAAAAAAAEAfAAAAAAAAyAAAAAAAAAAAZOgDAAAAAAAAAEvQBwAAAAAAAAAyuAsAAAAAAAAAGaAPAAAAAAAAAAD///////8PAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAAAABAAAAAAAAAKwNAAAAAAAAyAAAAAAAAAABZOgDAAAAAAAAAUvQBwAAAAAAAAEyuAsAAAAAAAAAMqAPAAAAAAAAABn
can't find the value 0x0d000000. Can you give me a url about the Toledo's solution? Thank you very much~

**Comment by shixuev5** [comment]
ow! I find the teloda solution detail in http://www.tonymacx86.com/mavericks-desktop-support/128926-mavericks-native-cpu-igpu-power-management-90.html#post1165500  #896

**Comment by shixuev5** [comment]
open the plist with Xcode the data view is correct. The problem has been solved . I will try clover kextsToPath. Thanks you.

**Comment by thePsguy** [comment]
How low does it go? After making the change, mine only goes as low as 1.0 GHz, not a point below it. Any Ideas?

**Comment by thePsguy** [comment]
According to ark.intel.com, my processor(5500U) has a low frequency of 600. Setting freq vector to 0x06000000 in Mac-937CB26E2E02BB01.plist does not help, even with the CPUFloor field set to 600.
The CPU goes to a minimum of 800MHz.

**Comment by Piker-Alpha** [comment]
That is the: "Configurable TDP-down Frequency". An extra low power mode with a lower TDP and LFM. Not the normal TDP and LFM frequency. Note the "Configurable" part. Only be used when the BIOS or driver activates it. Which as far as I know is not the case.

What I did notice is that we had some errors in Broadwell.cfg and thus it will generate a SSDT with a LFM of 500 MHz. Let me correct that right now... Done!

**Comment by thePsguy** [comment]
Oh okay. I see you have a script that fixes the frequency vector data. If I create a pull request, with a modification to this script, asking ,in case of affected boards, to download and execute that script, would a merge be considered?

**Comment by Piker-Alpha** [comment]
That depends on what you like to add/change.

Right now we only show an error message when the FrequencyVector data is missing, but you like to add a warning message to inform people that they should download/run freqVectorsEdit.sh Correct?

Are all Broadwell models affected by this?

**Comment by Lacedaemon** [comment]
Hey Pike, I'm a big fan of your work!

> The CPUFloor frequency can be found in the FrequencyVector data in Mac-937CB26E2E02BB01.plist The second or third value is set to 0x0d000000 (sorry cannot recall which one it is) and when you change that, say to 0x08000000, then it will throttle down to 800MHz.

Thanks, this worked for my i5-5200U on 10.11.3 w/ MacBookPro12,1 SMBIOS.

I have two other issues:
- I can reach all the CPU P-States except the final one, the 27 P-State.

**Without SSDT and `DropOem=true` in Clover:**

```
[ 10 11 12 (13) 14 15 16 17 18 19 20 21 22 23 24 25 26 27 ] iGPU P-States [ (6) ]
```

**With SSDT and your fix:**

```
[ (8) 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 ] iGPU P-States [ 10 11 (12) 13 16 17 ]
```
- My HD 5500 [won't throttle down to 300MHz](http://imgur.com/GH8UOwZ.jpg), when it should, according to [CPU-World](http://www.cpu-world.com/CPUs/Core_i5/Intel-Core%20i5-5200U%20Mobile%20processor.html).

Can a similar procedure be used to allow the GPU to throttle down?  Thanks!

**Comment by Piker-Alpha** [comment]
Thanks! 

You should only see the 27 multiplier at boot time. Not otherwise. That would be wrong.

And yes you can do the same, well sort of, for the IGPU but that is done in AppleGraphicsPowerManagement.kext/C*/Info.plist 
Look for the GT2Floor and GT3Floor (Number) values. Add them when missing. If you see 8 that is 400MHz. If you see 17 that is 750MHz. You want 6 for 300MHz.

**Comment by Lacedaemon** [comment]
> Look for the GT3Floor and GT3Floor (Number) values.

Did you mean `GT2Floor` and `GT3Floor`?  Because I only see one `GT3Floor` for my board-id (Mac-E43C1C25D4880AD6):

```
<key>GT2Floor</key>
<integer>8</integer>
<key>GT3Floor</key>
<integer>10</integer>
```

According to [this](https://www.firewolf.science/2015/04/framebuffer-data-extracted-from-appleintelbdwgraphicsframebuffer-binary/), HD 5500 is GT2 (not even sure if that applies here, forgive me if it doesn't).  So, would I change `GT2Floor`, `GT3Floor`, or both?

Thanks Pike!

**Edit:** It seems that 500MHz is actually normal... when my battery is depleted, it goes into LFM and _then_ the GPU dips to 300MHz.  Would be nice if you could confirm!

**Comment by Piker-Alpha** [comment]
Yes. That should have read GT2Floor and GT3Floor, but the latter one appears to be what Apple is checking since you have a value of 10 in it, and it throttles down to 500MHz. You could verify this by increasing this value to say 12 and see if it won't go below 600MHz.

**Comment by Lacedaemon** [comment]
Setting `GT3Floor` to 6 worked.  [Everything looks good now](http://imgur.com/HQBd6Mn.jpg).

Thanks for your help!

**Comment by Piker-Alpha** [comment]
This issue has been resolved with the release of v17.4

@thePsguy,

Thank you for the idea (credit added with Github commit).

See also: https://pikeralpha.wordpress.com/2016/01/29/beta-v17-4-ssdtprgen-sh-released/

**Comment by thePsguy** [comment]
Sorry, I got occupied. Thank you for the credit!
